sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -845,17 +845,33 @@ class SessionCatalog(
   }
 
   private def fromCatalogTable(metadata: CatalogTable, isTempView: Boolean): View = {
-    val viewText = metadata.viewText.getOrElse(sys.error("Invalid view without text."))
+    val viewText = metadata.viewText.getOrElse {
+      throw new IllegalStateException("Invalid view without text.")
+    }
     val viewConfigs = metadata.viewSQLConfigs
-    val viewPlan =
+    val parsedPlan =
       SQLConf.withExistingConf(View.effectiveSQLConf(viewConfigs, isTempView = isTempView)) {
         parser.parsePlan(viewText)
       }
-    View(
-      desc = metadata,
-      isTempView = isTempView,
-      output = metadata.schema.toAttributes,
-      child = viewPlan)
+    val viewColumnNames = metadata.viewQueryColumnNames
+    val viewPlan = if (viewColumnNames.nonEmpty) {
+      assert(viewColumnNames.length == metadata.schema.length)
+      // For view queries like `SELECT * FROM t`, the schema of the referenced table/view may
+      // change after the view has been created. We need to add an extra SELECT to pick the columns
```

Comment:
> For view queries like `SELECT * FROM t`, the schema of the referenced table/view may  change after the view has been created.

We already have some tests for the case above somewhere?

Reply:
I think so, the comment is copied from the previous code.

Reply:
Ah, ok.

Reply:
Hm? Is the comment "For view queries like `SELECT * FROM t`..." copied in this PR? I don't see its original place here.

This code seems copied from `EliminateView`, but its original comment is different. The `EliminateView`'s comment is more about resolution of attribute of view text.

Reply:
It's from https://github.com/apache/spark/pull/31368/files#diff-782f0d0b0d5fa6cf642285962eb0c831d9807e3f9ec2810f964292da89547e1aL38

I changed it a little bit to match the current context.